src/global_scheduler/global_scheduler.h

```diff
@@ -8,6 +8,11 @@
 #include "utarray.h"
 #include "uthash.h"
 
+/* The frequency with which the global scheduer checks if there are any tasks
+ * that were impossible to schedule at the time they arrived but can now be
+ * scheduled because new nodes have joined the cluster. */
+#define GLOBAL_SCHEDULER_TASK_CLEANUP_MILLISECONDS 1000
```

Comment:
100ms was arbitrary. If it's just for impossible tasks it doesn't need to be very frequent because the only time it is relevant is when a new node joins. However, if it's a more general "pending task queue" then maybe more frequently makes sense.